test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -417,27 +417,28 @@ static void ipsec_pkt_auth_err_set(odp_packet_t pkt)
 	odp_packet_copy_from_mem(pkt, len - sizeof(data), sizeof(data), &data);
 }
 
-static void ipsec_check_out_in_one(const ipsec_test_part *part,
+static void ipsec_check_out_in_one(const ipsec_test_part *part_outbound,
+				   const ipsec_test_part *part_inbound,
 				   odp_ipsec_sa_t sa,
 				   odp_ipsec_sa_t sa_in)
 {
-	int num_out = part->num_pkt;
+	int num_out = part_outbound->num_pkt;
 	odp_packet_t pkto[num_out];
 	int i;
 
-	num_out = ipsec_check_out(part, sa, pkto);
+	num_out = ipsec_check_out(part_outbound, sa, pkto);
 
 	for (i = 0; i < num_out; i++) {
-		ipsec_test_part part_in = *part;
+		ipsec_test_part part_in = *part_inbound;
```

Comment:
We can remove this if we pass part_inbound not as const, right?

Reply:
Yes, but I do not think it is good to make it non-const and have this function modify the parameter structure owned by the caller for no good reason. IOW, from the caller point-of-view it makes more sense to have a const pointer parameter since the function is not supposed to really return anything through it.

Reply:
Agreed. Thanks for the confirmation.